File: core/src/test/java/com/larryhsiao/nyx/core/jots/QueriedJotsTest.java
package com.larryhsiao.nyx.core.jots;

import com.silverhetch.clotho.Source;
import com.silverhetch.clotho.database.h2.MemoryH2Conn;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.sql.Connection;
import java.util.List;

/**
 * Unit-test for the class {@link QueriedJots}
 */
public class QueriedJotsTest {

    /**
     * Check created time exist
     */
    @Test
    public void createdTimeExist() throws Exception {
        Source<Connection> db = new JotsDb(new MemoryH2Conn());
        new NewJot(db, "content").value();
        List<Jot> jots = new QueriedJots(new AllJots(db)).value();
        Assertions.assertNotEquals(
                0,
                jots.get(0).createdTime()
        );
    }
}

File: core/src/test/java/com/larryhsiao/nyx/core/jots/QueriedJotsTest.kt
package com.larryhsiao.nyx.core.jots

import com.larryhsiao.clotho.Source
import com.larryhsiao.clotho.database.h2.MemoryH2Conn
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.sql.Connection

/**
 * Unit-test for the class [QueriedJots]
 */
class QueriedJotsTest {
    /**
     * Check created time exist
     */
    @Test
    @Throws(Exception::class)
    fun createdTimeExist() {
        val db: Source<Connection> = JotsDb(MemoryH2Conn())
        NewJot(db, "title", "content").value()
        val jots = QueriedJots(AllJots(db)).value()
        Assertions.assertNotEquals(
            0,
            jots[0].createdTime()
        )
    }
}